mkdocs/utils/__init__.py

```diff
@@ -222,13 +230,7 @@ def is_markdown_file(path):
     http://superuser.com/questions/249436/file-extension-for-markdown-files
     """
     ext = os.path.splitext(path)[1].lower()
-    return ext in [
-        '.markdown',
-        '.mdown',
-        '.mkdn',
-        '.mkd',
-        '.md',
-    ]
+    return any(fnmatch.fnmatch(ext, pattern) for pattern in markdown_patterns)
```

Comment:
I like this, but I don't think we need to break out the extension from the path. Just pass `path` to `fnmatch` and remove line 232.

Reply:
Or, leave it unchanged as we have a list of extensions, not patterns. It probably should come down to which is more performant. If `fnmatch` fast enough, then converted the extensions to pattern for each call might not matter. Otherwise, don't use `fnmatch` here.